src/IIIFPresentation/Services/Manifests/ManifestMerger.cs

```diff
@@ -87,11 +87,12 @@ private Dictionary<AssetId, Canvas> BuildAssetIdToCanvasLookup(Manifest namedQue
         {
             return namedQueryManifest
                 .Items!
-                .ToDictionary(canvas => canvas.GetAssetIdFromNamedQueryCanvasId(), canvas => canvas);
+                .GroupBy(canvas => canvas.GetAssetIdFromNamedQueryCanvasId())
+                .ToDictionary(canvas => canvas.Key, canvas => canvas.First());
```

Comment:
Not sure this is required? This would only be required if the NQ is returning > 1 separate canvases for the same Asset, which shouldn't be possible. The "duplicate" Asset is only a Presentation concern, Protagonist doesn't have any notion of that - it just has a single Asset.